src/check.c

```diff
@@ -149,10 +150,49 @@ TCase *tcase_create(const char *name)
     tc->ch_sflst = check_list_create();
     tc->unch_tflst = check_list_create();
     tc->ch_tflst = check_list_create();
+    tc->tags = NULL;
```

Comment:
The other lists are created and destroyed unconditionally, Why not follow this pattern? Or, is there an advantage to doing it this way?